### ISSUE TITLE
Please document: What is this for?

### ISSUE THREAD

**Issue by probonopd** [vioken/qwlroots#312]
Hello, I am looking for a way to write a minimalistic tiling window manager using wlroots that uses Qt6 Widgets to draw window decorations (title, close button, minimize/maximize buttons). I want it to support all Wayland protocols that wlroots is supporting without having much work on my side every time a new protocol is added.

Can this be done using qwlroots? Is there even an example maybe?

**Comment by Dami-star** [comment]
You may need to rely on waylib. qwlroots is just a Qt-style binding of wlroots, which simplifies the way to call wlroots API in Qt projects.
https://github.com/vioken/waylib

**Comment by probonopd** [comment]
Thanks, but isn't wlroots for making a Wayland compositor?

**Comment by zccrs** [comment]
> Thanks, but isn't wlroots for making a Wayland compositor?

Yes. you can use qwlroots like as the wlroots to make a Wayland compositor, but you can't using QtWidgets with qwlroots. https://github.com/vioken/qwlroots/blob/master/examples/tinywl/main.cpp this is an example of qwlroots, you can using QtCore with qwlroots, eg: use the Qt style signals to connect the wayland signals.

For the Waylib, you can't use QtWidgets also, but you can  using QtDeclarative with it, https://github.com/vioken/waylib/tree/master/examples/tinywl this is an example for use QtQuick with wlroots.
